packages/gatsby-plugin-page-creator/src/gatsby-node.js

```diff
@@ -48,48 +49,56 @@ exports.createPagesStatefully = async (
     )
   }
 
-  const pagesDirectory = systemPath.posix.join(pagesPath)
-  const pagesGlob = `${pagesDirectory}/**/*.{${exts}}`
+  const pagesDirectory = systemPath.resolve(process.cwd(), pagesPath)
+  const pagesGlob = `**/*.{${exts}}`
 
   // Get initial list of files.
-  let files = await glob(pagesGlob)
-  files.forEach(file => _createPage(file, pagesDirectory, createPage))
+  let files = await glob(pagesGlob, { cwd: pagesPath })
+  files.forEach(file => _createPage(file, pagesDirectory, createPage, ignore))
 
   // Listen for new component pages to be added or removed.
   chokidar
-    .watch(pagesGlob)
+    .watch(pagesGlob, { cwd: pagesPath })
     .on(`add`, path => {
+      path = slash(path)
       if (!_.includes(files, path)) {
-        _createPage(path, pagesDirectory, createPage)
+        _createPage(path, pagesDirectory, createPage, ignore)
         files.push(path)
       }
     })
     .on(`unlink`, path => {
       path = slash(path)
       // Delete the page for the now deleted component.
       store.getState().pages.forEach(page => {
-        if (page.component === path) {
+        const componentPath = systemPath.join(pagesDirectory, path)
```

Comment:
I'm not sure but should we go for posix.join? I'm unsure if it's needed here
```suggestion
        const componentPath = systemPath.posix.join(pagesDirectory, path)
```

Reply:
The [createPages API](https://www.gatsbyjs.org/docs/actions/#createPage) example doesn't use posix for resolving component path, so I figure it's best to let Node use the native OS syntax for finding the component files.

```js
createPage({
  path: `/my-sweet-new-page/`,
  component: path.resolve(`./src/templates/my-sweet-new-page.js`),
  // The context is passed as props to the component as well
  // as into the component's GraphQL query.
  context: {
    id: `123456`,
  },
})
```